inc/impact.class.php

```diff
@@ -0,0 +1,1091 @@
+
+   public static function getTypeName($nb = 0) {
+      return _n('Asset impact', 'Asset impacts', $nb);
+   }
+
+   public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0) {
+      global $CFG_GLPI, $DB;
+
+      // Class of the current item
+      $class = get_class($item);
+
+      // Only enabled for CommonDBTM
+      if (!is_a($item, "CommonDBTM")) {
+         throw new InvalidArgumentException(
+            "Argument \$item ($class) must be a CommonDBTM)."
+         );
+      }
+
+      // If on an asset, get the number its direct dependencies
+      if (isset($CFG_GLPI['impact_asset_types'][$class])) {
+         $total = count($DB->request([
+            'FROM'   => ImpactRelation::getTable(),
+            'WHERE'  => [
+               'OR' => [
+                  [
+                     'itemtype_source' => get_class($item),
+                     'items_id_source' => $item->getID(),
+                  ],
+                  [
+                     'itemtype_impacted' => get_class($item),
+                     'items_id_impacted' => $item->getID(),
+                  ]
+               ]
+            ]
+         ]));
+      } else if (is_a($item, "CommonITILObject")) {
+         // Tab name for an ITIL object : always 0
+         $total = 0;
+      } else {
+         // Neither an asset nor an CommonITILObject
+         throw new InvalidArgumentException(
+            "Argument \$item ($class) is not a valid target for impact analysis."
+         );
+      }
+
+      return self::createTabEntry(
+         _n('Impact ', 'Impacts ', Session::getPluralNumber()),
+         $total
+      );
+   }
+
+   public static function displayTabContentForItem(
+      CommonGLPI $item,
+      $tabnum = 1,
+      $withtemplate = 0
+   ) {
+      global $CFG_GLPI;
+
+      $class = get_class($item);
+
+      // Only enabled for CommonDBTM
+      if (!is_a($item, "CommonDBTM")) {
+         throw new InvalidArgumentException(
+            "Argument \$item ($class) must be a CommonDBTM)."
+         );
+      }
+
+      $ID = $item->getID();
+
+      // Don't show the impact analysis on new object
+      if ($item->isNewID($ID)) {
+         return false;
+      }
+
+      // Check READ rights
+      $itemtype = $item->getType();
+      if (!$itemtype::canView()) {
+         return false;
+      }
+
+      // For an ITIL object, load the first linked element by default
+      if (is_a($item, "CommonITILObject")) {
+         $linked_items = $item->getLinkedItems();
+
+         // Search for a valid linked item of this ITILObject
+         $found = false;
+         foreach ($linked_items as $linked_item) {
+            $class = $linked_item['itemtype'];
+            if (isset($CFG_GLPI['impact_asset_types'][$class])) {
+               self::printAssetSelectionForm($linked_items);
+               $found = true;
+               $item = new $class;
+               $item->getFromDB($linked_item['items_id']);
```

Comment:
`$found = true` should be set only if `getFromDB` is true aswell

Reply:
Fixed in latest commit.